modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/report/BaseDDMFormFieldTypeReportProcessor.java

```diff
@@ -0,0 +1,176 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.internal.report;
+
+import com.liferay.dynamic.data.mapping.form.field.type.DDMFormFieldTypeServicesTracker;
+import com.liferay.dynamic.data.mapping.model.DDMFormFieldOptions;
+import com.liferay.dynamic.data.mapping.model.Value;
+import com.liferay.dynamic.data.mapping.report.DDMFormFieldTypeReportProcessor;
+import com.liferay.dynamic.data.mapping.storage.DDMFormFieldValue;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+
+import java.util.Map;
+import java.util.stream.Stream;
+
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Cleydyr de Albuquerque
+ */
+public abstract class BaseDDMFormFieldTypeReportProcessor
+	implements DDMFormFieldTypeReportProcessor {
+
+	public JSONObject process(
+			DDMFormFieldValue ddmFormFieldValue, JSONObject fieldJSONObject,
+			long formInstanceRecordId, String ddmFormInstanceReportEvent)
+		throws Exception {
+
+		doProcess(
+			ddmFormFieldValue, fieldJSONObject, formInstanceRecordId,
+			ddmFormInstanceReportEvent);
+
+		Map<String, Object> ddmFormFieldTypeProperties =
+			ddmFormFieldTypeServicesTracker.getDDMFormFieldTypeProperties(
+				ddmFormFieldValue.getType());
+
+		fieldJSONObject.put(
+			"icon", ddmFormFieldTypeProperties.get("ddm.form.field.type.icon")
+		).put(
+			"title", ddmFormFieldTypeProperties.get("ddm.form.field.type.label")
+		).put(
+			"chartComponentName", getChartComponentName()
+		).put(
+			"chartComponentProps",
+			getChartComponentPropsJSONObject(fieldJSONObject, ddmFormFieldValue)
```

Comment:
@cleydyr Let’s keep the parameters in the same order as they are in the `process` methodo, i.e., `ddmFormFieldValue` before `fieldJSONObject`. Thanks!